src/Net/LinkShortener.php

```diff
@@ -13,8 +13,9 @@ protected static function GetRedisInstance(): Redis {
         return $redis;
     }
 
-    protected static function GenerateShortString(): string {
-        return bin2hex(random_bytes(8));
+    protected static function GenerateShortString(string $longUrl): string {
+        $hash = substr(md5($longurl . time()), 0, 8);
```

Comment:
Don't hash with time. The intent is that the same $longUrl will return the same short link. Also make it length 16, less chance of hash collisions that way.
```suggestion
        $hash = substr(md5($longurl), 0, 16);
```